buildSrc/src/main/java/org/elasticsearch/gradle/Version.java

```diff
@@ -0,0 +1,183 @@
+                "Invalid version format: '" + s + "'. Should be major.minor.revision[-(alpha|beta|rc)Number][-SNAPSHOT]"
+            );
+        }
+
+        return new Version(
+                Integer.parseInt(matcher.group(1)),
+                parseSuffixNumber(matcher.group(2)),
+                parseSuffixNumber(matcher.group(3)),
+                matcher.group(4),
+                matcher.group(5) != null
+        );
+    }
+
+    @Override
+    public String toString() {
+        final String snapshotStr = snapshot ? "-SNAPSHOT" : "";
+        return String.valueOf(getMajor()) + "." + String.valueOf(getMinor()) + "." + String.valueOf(getRevision()) +
+                (suffix == null ? "" : suffix) + snapshotStr;
+    }
+
+    public boolean before(Version compareTo) {
+        return id < compareTo.getId();
+    }
+
+    public boolean before(String compareTo) {
+        return before(fromString(compareTo));
+    }
+
+    public boolean onOrBefore(Version compareTo) {
+        return id <= compareTo.getId();
+    }
+
+    public boolean onOrBefore(String compareTo) {
+        return onOrBefore(fromString(compareTo));
+    }
+
+    public boolean onOrAfter(Version compareTo) {
+        return id >= compareTo.getId();
+    }
+
+    public boolean onOrAfter(String compareTo) {
+        return onOrAfter(fromString(compareTo));
+    }
+
+    public boolean after(Version compareTo) {
+        return id > compareTo.getId();
+    }
+
+    public boolean after(String compareTo) {
+        return after(fromString(compareTo));
+    }
+
+    public boolean onOrBeforeIncludingSuffix(Version otherVersion) {
+        if (id != otherVersion.getId()) {
+            return id < otherVersion.getId();
+        }
+
+        if (suffix.equals("")) {
+            return otherVersion.getSuffix().equals("");
+        }
+
+
+        return otherVersion.getSuffix().equals("") || suffix.compareTo(otherVersion.getSuffix()) < 0;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Version version = (Version) o;
+        return major == version.major &&
+                minor == version.minor &&
+                revision == version.revision &&
+                id == version.id &&
+                snapshot == version.snapshot &&
+                Objects.equals(suffix, version.suffix);
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(major, minor, revision, id, snapshot, suffix);
+    }
+
+    public final int getMajor() {
+        return major;
+    }
+
+    public final int getMinor() {
+        return minor;
+    }
+
+    public final int getRevision() {
+        return revision;
+    }
+
+    protected final int getId() {
+        return id;
+    }
+
+    public final boolean getSnapshot() {
+        return snapshot;
+    }
+
+    public final boolean isSnapshot() {
```

Comment:
Why do you have `get` and `is`? If we don't make these public final members, then at least there should only be one of these methods.

Reply:
It's just an oversight. I'm removing `get`